src/ServiceInsight/Framework/WorkNotifier.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace ServiceInsight.Framework
+{
+    using System;
+    using System.Reactive.Disposables;
+    using Caliburn.Micro;
+    using ServiceInsight.Framework.Events;
+
+    public interface IWorkNotifier
+    {
+        IDisposable NotifyOfWork();
+
+        IDisposable NotifyOfWork(string startMessage);
+
+        IDisposable NotifyOfWork(string startMessage, string finishMessage);
+    }
+
+    class WorkNotifier : IWorkNotifier
+    {
+        IEventAggregator eventAggregator;
+
```

Comment:
Not all WorkTrackers are WorkNotifiers, and vice versa. One is a publisher, the other is a subscriber.